pymeasure/instruments/instrument.py

```diff
@@ -140,43 +141,53 @@ def control(get_command, set_command, docs,
         :param check_get_errors: Toggles checking errors after getting
         """
 
+        inverse = None
         if map_values and isinstance(values, dict):
             # Prepare the inverse values for performance
             inverse = {v: k for k, v in values.items()}
 
         def fget(self):
+            values_local = values
+            inverse_local = inverse
             vals = self.values(get_command, **kwargs)
             if check_get_errors:
                 self.check_errors()
+            if isinstance(values_local, str):
+                values_local = getattr(self, values_local)
             if len(vals) == 1:
                 value = get_process(vals[0])
                 if not map_values:
                     return value
-                elif isinstance(values, (list, tuple, range)):
-                    return values[int(value)]
-                elif isinstance(values, dict):
-                    return inverse[value]
+                elif isinstance(values_local, (list, tuple, range)):
+                    return values_local[int(value)]
+                elif isinstance(values_local, dict):
+                    if inverse_local is None:
+                        inverse_local = {v: k for k, v in values_local.items()}
```

Comment:
it is weird to do this inverse creation twice (here and above) - I would rather prefer we adapt the logic condition above, or only do it here if needed, but not in both places.

Reply:
I would agree, but the reason for this duplicate is performance: if an actual dict is provided as values it is possible to define te inverse beforehand, which means it is only defined once. In case of a string, it has to be calculated every single time, because the values can change.

Reply:
Agreed, makes sense.